misc/test/go.mod

```diff
@@ -1,11 +1,13 @@
 module github.com/pulumi/examples/misc/test
 
-go 1.20
```

Comment:
go1.20 EOL'd in Feb 6, 2024. Should be OK? Maybe merge and fix forward?

Reply:
looks like the tests are happy, so I see no reason not to bump to 1.21